GSdx-d3d: OSD port to d3d from gl

lightningterror opened the pull request.

lightningterror committed:
GSdx-d3d: OSD port from gl.

lightningterror committed:
Correct Blend maybe ?

tadanokojin committed:
GSdx-d3d: Update some OSD ogl functions to their dx counterparts

tadanokojin committed:
GSdx-d3d: Enable OSD in DX11 settings.

lightningterror commented on plugins/GSdx/GSSettingsDlg.cpp:
```diff
@@ -435,7 +435,7 @@ void GSSettingsDlg::UpdateControls()
 
 		// OSD Configuration
-		EnableWindow(GetDlgItem(m_hWnd, IDC_OSDBUTTON), ogl);
+		EnableWindow(GetDlgItem(m_hWnd, IDC_OSDBUTTON), dx11 || ogl);
```
Note to self, change this in the future to !dx9.

tadanokojin commented:
`m_font` is working as expected as far as I can tell:

OGL
![font_atlas_ogl](https://user-images.githubusercontent.com/26046960/47914777-2c451700-de77-11e8-84db-8a0d956d2ff1.png)

DX
![font_atlas_dx](https://user-images.githubusercontent.com/26046960/47914785-3404bb80-de77-11e8-8556-9b62340ab090.png)

`POSITION0` looks good as well
![001668](https://user-images.githubusercontent.com/26046960/47915290-c22d7180-de78-11e8-8f55-a0b62e1ff2f1.png)

Not so confident about `TEXCOORD0` unfortunately I don't have a nice visualization to compare with OGL.
![001669](https://user-images.githubusercontent.com/26046960/47915317-d70a0500-de78-11e8-87e6-d6e7972084d3.png)

In any case, the draw happens and the vertex data is making it to the IA, but it's not making it to the OM.

Looks like the issue is somewhere in the pipeline.

I have a small change that makes `m_font` appear as it does OGL so I'll push with that soon. Also in it's current state `RenderOsd` is not being called because no `bs` is being passed by the caller so we're just dumping into an empty virtual like before.

I did try a few things. 
1. Disabling blend state and the shader in OGL will still render.
2. Passing some of the blend states available to the class didn't change much.
3. Comparing `StretchRect` and the like in OGL showed that DX has some extra pipeline steps. None of these has any effect though.
4. I've tried changing various things in the pipeline but not seeing any results there.

For now I've been familiarizing myself with the DX pipeline.

lightningterror commented:
I guess you're almost there then

tadanokojin committed:
GSdx-d3d: Updates to osd pipeline and texture fix

- use DXGI_FORMAT_R8_UNORM to make m_font match OGL
- use m_merge blend state for alpha blending
- add some additional steps found by comparing some OGL methods to DX
